### ISSUE TITLE
Pages not calling onPageSelected

### ISSUE THREAD

**Issue by qavan** [arkivanov/Decompose#526]
Pages not calling onPageSelected, for example, if swiped slowly to page

Decompose: 2.1.4

```
sealed class PageConfig : Parcelable {
    @Parcelize
    data object Page1: PageConfig()
    @Parcelize
    data object Page2: PageConfig()
}

class PageComponent(val page: PageConfig)

@ExperimentalDecomposeApi
class PagesComponent(context: ComponentContext): ComponentContext by context {
    private val navigation = PagesNavigation<PageConfig>()

    val tabs: Value<ChildPages<PageConfig, PageComponent>> = childPages(
        source = navigation,
        initialPages = {
            Pages(
                items = listOf(
                    PageConfig.Page1,
                    PageConfig.Page2,
                ),
                0
            )
        },
    ) { config, _ ->
        return@childPages PageComponent(config)
    }

    fun selectPage(page: Int) {
        navigation.select(page)
    }
}

@ExperimentalDecomposeApi
@Preview
@Composable
fun PagesIssue() {
    val lifecycleOwner = LocalLifecycleOwner.current
    val component = remember { PagesComponent(DefaultComponentContext(lifecycleOwner.lifecycle.asEssentyLifecycle())) }
    val pages by component.tabs.subscribeAsState()
    Column {
        Text(
            text = "${pages.selectedIndex}",
        )
        Pages(
            pages = component.tabs,
            modifier = Modifier.weight(1f),
            onPageSelected = component::selectPage
        ) { index, page ->
            Text(
                text = "${page.page}",
                modifier = Modifier
                    .fillMaxSize()
                    .wrapContentSize(align = Alignment.TopCenter)
                    .padding(top = 12.dp)
            )
        }
    }
}
```

https://github.com/arkivanov/Decompose/assets/36155897/e79fb7c9-ca53-4fad-9bbc-5564e4bd36d7

**Comment by arkivanov** [comment]
Hello. I believe this should be fixed in Compose 1.6.0-alpha08. Please try this version.

**Comment by qavan** [comment]
yeah, fixed(checked in 1.6.0-beta01)
